src/lib/util.ts

```diff
@@ -50,8 +52,32 @@ export default class Util {
   public static getWeb3() {
     return getArc().web3;
   }
-  public static async getNetworkId() {
-    return await getArc().web3.eth.net.getId();
+
+  /**
+   * return network id, independent of the presence of Arc
+   * @param web3Provider
+   */
+  public static async getNetworkId(web3Provider?: any): Promise<string> {
+    let arc: any;
+    let web3: any;
+
+    try {
+      arc = getArc();
+    } catch {}
+
+    if (arc) {
+      web3 = arc.web3;
+    } else if ((<any> window).web3) {
+      web3 = (<any> window).web3;
+    } else if (web3Provider) {
+      web3 = new Web3(web3Provider);
+    }
+
+    if (!web3) {
+      throw new Error("getNetworkId: unable to find web3");
+    }
+
+    return (await (web3.eth.net ? web3.eth.net.getId() : promisify(web3.version.getNetwork)())).toString();
```

Comment:
O don't think promsify is needed here, as the web3 functions should return promises (if you do not provide a callback)

Reply:
pre-1.0 we need promisify

Reply:
That is, one has to provide a callback or else in some cases (I forget now which) these functions will crash because the don't support async operations.

Reply:
https://ethereum.stackexchange.com/questions/47366/the-metamask-web3-object-does-not-support-synchronous-methods-like-personal-newa

Reply:
true, this is the injected old web3... bleah.